typedmodels/models.py

```diff
@@ -410,21 +410,21 @@ def recast(self, typ=None):
                     # Instead, we raise an error when the object is saved.
                     return
             else:
-                self.type = self._typedmodels_type
+                self.dtm_type = self._typedmodels_type
 
         if typ is None:
-            typ = self.type
+            typ = self.dtm_type
```

Comment:
Again I think this does a db query, unless it was just set above.

Is that true? I can't find anything suggesting that Django uses the cache for foreignkeys to ContentType.

```suggestion
            typ = ContentType.objects.get_for_id(self.dtm_type_id)```